django/applications/catmaid/control/skeletonexport.py

```diff
@@ -1164,19 +1164,18 @@ def __init__(self, parent_id, x, y, z):
     class Skeleton():
         def __init__(self):
             self.nodes = {}  # type: Dict[Any, Node]
-            self.raw_cable = 0
-            self.smooth_cable = 0
-            self.principal_branch_cable = 0
+            self.raw_cable = 0.0
+            self.smooth_cable = 0.0
+            self.principal_branch_cable = 0.0
             self.n_ends = 0
             self.n_branch = 0
             self.n_pre = 0
             self.n_post = 0
 
-    skeletons = defaultdict(dict)  # type: DefaultDict[Any, Dict]
-                                   # skeleton ID vs (node ID vs Node)
+    skeletons = {} # type: Dict[Any, Skeleton]
+                   # skeleton ID vs (node ID vs Node)
     for row in cursor.fetchall():
-        skeleton = skeletons.get(row[2])
-        if not skeleton:
+        if row[2] not in skeletons:
```

Comment:
While this check certainly matches the previous `if not skeleton`, where is the default/else case? In other words, assuming nodes are unordered: if the skeleton of a node has already been seen, how will the `skeleton` variable be updated to reference the already existing `Skeleton` object stored in the `skeletons` `dict`. I could see this change work when replacing `skeleton` in L1181 with e.g. `skeletons[row[2]]`.

Reply:
Ah, deleted a little too much. Will revise.

Reply:
:+1: Didn't see the updated code, sorry.